OpenRA.Mods.Common/Traits/World/ActorMap.cs

```diff
@@ -285,51 +286,62 @@ public bool HasFreeSubCell(CPos cell, bool checkTransient = true)
 
 		public SubCell FreeSubCell(CPos cell, SubCell preferredSubCell = SubCell.Any, bool checkTransient = true)
 		{
-			if (preferredSubCell != SubCell.Any && !AnyActorsAt(cell, preferredSubCell, checkTransient))
+			var uv = cell.ToMPos(map);
+			if (!influence.Contains(uv))
+				return SubCell.Invalid;
+
+			if (preferredSubCell != SubCell.Any && !AnyActorsAt(uv, cell, preferredSubCell, checkTransient))
 				return preferredSubCell;
 
-			if (!AnyActorsAt(cell))
+			if (!AnyActorsAt(uv, cell.Layer))
 				return map.Grid.DefaultSubCell;
 
 			for (var i = (int)SubCell.First; i < map.Grid.SubCellOffsets.Length; i++)
-				if (i != (int)preferredSubCell && !AnyActorsAt(cell, (SubCell)i, checkTransient))
+				if (i != (int)preferredSubCell && !AnyActorsAt(uv, cell, (SubCell)i, checkTransient))
 					return (SubCell)i;
 
 			return SubCell.Invalid;
 		}
 
 		public SubCell FreeSubCell(CPos cell, SubCell preferredSubCell, Func<Actor, bool> checkIfBlocker)
 		{
-			if (preferredSubCell != SubCell.Any && !AnyActorsAt(cell, preferredSubCell, checkIfBlocker))
+			var uv = cell.ToMPos(map);
+			if (!influence.Contains(uv))
+				return SubCell.Invalid;
```

Comment:
This looks like a behaviour change. Beforehand `AnyActorsAt` would return false and then `map.Grid.DefaultSubCell` (or `preferredSubCell`) would be returned and not `SubCell.Invalid`?

Reply:
True. I now realize that `Invalid` is never the correct return value - but `preferredCell` or `FirstCell`. 

When a cell is not on the map - a sub cell is returned  - either the preferred or the first. The caller will not know it passed an off map cell. It might be better to return `Invalid` perhaps to allow the caller to cancel further processing. I.e. path planning may now continue longer than needed?